### PR TITLE
feat: add --keep-alive-timeout CLI arg to adjust server connection timeout

### PR DESCRIPTION
**Pull request by jlipps**
This is the Appium server side of https://github.com/appium/appium-base-driver/pull/443

### CODE REVIEW

File: lib/parser.js [hunk 1]
```diff
@@ -66,6 +66,15 @@ const args = [
           'on this server'
   }],
 
+  [['-ct', '--connection-timeout'], {
```
Comment (mykola-mokhnach) [inline lib/parser.js:69]: this is actually --keep-alive-timeout. Lets not mix different entities

File: lib/parser.js [hunk 2]
```diff
@@ -66,6 +66,15 @@ const args = [
...
+    required: false,
+    default: null,
+    dest: 'keepAliveTimeout',
+    type: 'int',
+    help: 'Number of seconds the Appium server should apply as the connection timeout for all ' +
```
Comment (mykola-mokhnach) [inline lib/parser.js:74]: We should mention the default value

Comment (mykola-mokhnach) [reply lib/parser.js:74]: connection timeout is something different. Keepalive works on the lower level than HTTP does (TCP one)

Comment (jlipps) [reply lib/parser.js:74]: we use the same value for both, in the code, though...?

**Comment by jlipps** [comment]
Updated to `--keep-alive-timeout`

**Review by KazuCocoa — APPROVED** [review]
The title should also be --keep-alive-timeout :)

File: lib/parser.js [hunk 3]
```diff
@@ -66,6 +66,15 @@ const args = [
           'on this server'
   }],
 
+  [['-ka', '--keep-alive-timeout'], {
+    required: false,
+    default: null,
+    dest: 'keepAliveTimeout',
+    type: 'int',
+    help: 'Number of seconds the Appium server should apply as the connection timeout for all ' +
```
Comment (mykola-mokhnach) [inline lib/parser.js:74]: https://stackoverflow.com/questions/49704708/what-is-a-connection-timeout-during-a-http-request

Although, keep alive is https://nodejs.org/api/http.html#http_server_keepalivetimeout

Comment (jlipps) [reply lib/parser.js:74]: @mykola-mokhnach I see. the main point of my change has to do with the connection timeout aspect. What would you want to see changed here to make things clear, if anything?

Comment (mykola-mokhnach) [reply lib/parser.js:74]: The code is fine. I just thought about tuning the help string to avoid possible misunderstandings

**Pull request merged**